### ISSUE TITLE
Progress bar for deploying functions?

### ISSUE THREAD

**Issue by donjae** [GoogleCloudPlatform/cloud-functions-emulator#76]
### Description

Feature request to have some sort of progress bar for deploying functions. Can't tell if functions is running into issues or just hanging

**Comment by donjae** [comment]
apologies, should have included it at the top:

node@v7.6.0
google-cloud/functions-emulator@1.0.0-alpha.15

Deployed the basic 'helloWorld' function, which ran fine initially after install (` functions deploy helloWorld --trigger-http`). But after deleting it and gracefully exiting it, (even grep node and closing the trailing instances to make sure nothing is running), I still have issues reinitializing and deploying functions until I restart my machine. But I digress...

Trying to deploy a real function on the emulator that has a few dependencies takes awhile, and it seems to be hanging without any sort of feedback. Would be nice if there was some sort of progress bar or [error] feedback if the emulator is running into something.

**Comment by jmdobry** [comment]
I recommend using Node.js v6.9.1, as that's the version of Node.js used by the production Cloud Functions service.

I'll still need more information in order to reproduce the problem you're seeing. Can you post here the code for a function that I can deploy in order to reproduce your issue? Can you post what your `package.json` file looks like? Can you post the output of `functions config list -j`?

**Comment by donjae** [comment]
`package.json` for the file: 

```
{
  "name": "cloud-function-image",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "nodemon ./index.js localhost 8081"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@google-cloud/storage": "^0.8.0",
    "fast-crc32c": "^1.0.4",
    "multer": "^2.0.0-alpha.6",
    "multiparty": "^4.1.3",
    "rethinkdbdash": "^2.3.28",
    "sharp": "^0.17.2",
    "shortid": "^2.2.8"
  },
  "devDependencies": {
    "nodemon": "^1.11.0"
  }
}
```


running the command `functions config list -j` 
```
{
  "debug": false,
  "debugPort": 5858,
  "grpcHost": "localhost",
  "grpcPort": 8009,
  "inspect": false,
  "inspectPort": 9229,
  "logFile": "logs/cloud-functions-emulator.log",
  "region": "us-central1",
  "restHost": "localhost",
  "restPort": 8008,
  "service": "rest",
  "storage": "configstore",
  "supervisorHost": "localhost",
  "supervisorPort": 8010,
  "timeout": 60000,
  "useMocks": false,
  "verbose": false,
  "projectId": "kappa-1268"
}
```

I am noticing a mismatch in ports for my functions code, and where the emulator is running, but not sure if that's the issue

**Comment by jmdobry** [comment]
@donjae Are you still encountering issues with deployment? Deploying to the Emulator should be really fast, no more than a couple seconds.

**Comment by donjae** [comment]
Yeah, still encountering issues. But like you mentioned earlier, I'm using a non-supported node version. 

I wish I could be more specific in what's causing the issue, but don't really have much info besides that it's stuck on loading without any response. I'll see if I can pull up some logs later when I try again

**Comment by jmdobry** [comment]
I'll be publishing another version soon that will hopefully solve your issue.

**Comment by donjae** [comment]
thanks fam, I'll keep an eye out and test it out when it's released and lyk

**Comment by jmdobry** [comment]
I released [1.0.0-alpha.16](https://github.com/GoogleCloudPlatform/cloud-functions-emulator/releases/tag/1.0.0-alpha.16), can you see if it works better for you?
